mypy_django_plugin/transformers/managers.py

```diff
@@ -104,17 +108,122 @@ def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[P
             # only needed for pluign-generated querysets.
             ret_type = Instance(queryset_info, [manager_instance.args[0], manager_instance.args[0]])
         variables = []
+    args_types = method_type.arg_types[1:]
+    if _has_compatible_type_vars(base_that_has_method):
```

Comment:
Can we move this check to where the manager and queryset are "fused" together?

https://github.com/typeddjango/django-stubs/blob/218d5fd1fb19dcc68811d3568b2f47b4ecaecabb/mypy_django_plugin/transformers/managers.py#L326-L329

That would be more efficient, since it'll only happen once, while this is called for each accessed manager/queryset method. We would also get better context if we'd emit an error message. Then we can only do replacement of type vars in here.

We should also ensure that upper bounds of any `TypeVar`s are compatible (i.e. `mypy.nodes.TypeVarExpr.upper_bound` or `mypy.types.TypeVarType.upper_bound`)

Reply:
Hm, thinking twice about this, it's quite branched off and self contained. It's also only new additions and shouldn't really break anything.

We could try to improve in subsequent PR:s instead, if you'd prefer that? But I'd say we at least remove `_get_base_containing_method` in favor of mypy's implementation.

Reply:
yes maybe let's do that in another PR.